backend/app/adapter/gqlapi/resolver/authmutation.go

```diff
@@ -22,51 +21,9 @@ type AuthMutation struct {
 	shortLinkUpdater shortlink.Updater
 }
 
-// ShortLinkInput represents possible ShortLink attributes
-type ShortLinkInput struct {
-	LongLink    *string
-	CustomAlias *string
-	ExpireAt    *time.Time
-}
-
-// TODO(#840): remove this business logic and move it to use cases
-func (s *ShortLinkInput) isEmpty() bool {
-	return *s == ShortLinkInput{}
-}
-
-// TODO(#840): remove this business logic and move it to use cases
-func (s *ShortLinkInput) longLink() string {
-	if s.LongLink == nil {
-		return ""
-	}
-	return *s.LongLink
-}
-
-// TODO(#840): remove this business logic and move it to use cases
-func (s *ShortLinkInput) customAlias() string {
-	if s.CustomAlias == nil {
-		return ""
-	}
-	return *s.CustomAlias
-}
-
-// TODO(#840): remove this business logic and move it to use cases
-func (s *ShortLinkInput) createUpdate() *entity.ShortLink {
-	if s.isEmpty() {
-		return nil
-	}
-
-	return &entity.ShortLink{
-		Alias:    s.customAlias(),
-		LongLink: s.longLink(),
-		ExpireAt: s.ExpireAt,
-	}
-
-}
-
 // CreateShortLinkArgs represents the possible parameters for CreateShortLink endpoint
 type CreateShortLinkArgs struct {
-	ShortLink ShortLinkInput
+	ShortLink entity.ShortLinkInput
```

Comment:
This is part of GraphQL definition. I recommend we define a seperate `ShortLinkInput` inside gql package and make it to `entity. ShortLinkInput` so that when we make changes to Graphql schema in the future, the rest of the system is left untouched.

Reply:
So if I understand correctly, I should just add a simple typedef that points the gql `ShortLInkInput` to the entity one. This is so that in the future, it can be easily swapped out for something else without major refactoring.

Reply:
@Coteh Exactly!

Reply:
Leaving a comment for myself, @byliuyang clarified over Slack that there should be a new type in `gqlapi` package that represents `shortLinkInput` within GraphQL schema, and instances of this type passed into API layer will be converted to `ShortLinkInput` from `entity` package before being passed into the use case repository methods. This will allow for use case methods to be called from other endpoints such as gRPC in the future.

Reply:
I will resolve this comment once this change has been made.

Reply:
And allow the APIs to evolve separately.